### PR TITLE
4.3 Patches

### PR DESCRIPTION
**Pull request by shyrkov**
Proposing several patches for the 4.3 IzPack branch we (http://www.jahia.com , https://github.com/Jahia) did for our project.

Thank you in advance for reviewing them!

### CODE REVIEW

File: src/lib/com/izforge/izpack/panels/ExtendedInstallPanelAutomationHelper.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
+/**
```
Comment (jponge) [inline src/lib/com/izforge/izpack/panels/ExtendedInstallPanelAutomationHelper.java:1]: We can't accept GPL code...

File: src/lib/com/izforge/izpack/panels/ExtendedInstallPanelConsoleHelper.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2002-2008 Jahia Ltd
+ *
+ * Licensed under the JAHIA COMMON DEVELOPMENT AND DISTRIBUTION LICENSE (JCDDL),
```
Comment (jponge) [inline src/lib/com/izforge/izpack/panels/ExtendedInstallPanelConsoleHelper.java]: It's ok to have (c) Jahia Ltd, but please have the code under the ASL 2 for consistency reasons :-)

**Comment by jponge** [comment]
Apart from the obvious licensing issues, your changes could make a nice new release of the 4.3 branch although we did not plan a further release.
- Can you fix the licensing problems?
- Would you merge/port the changes to the `master` branch?
- Would you update / contribute to the (new) documentation? http://docs.codehaus.org/display/IZPACK/User+documentation

**Comment by shyrkov** [comment]
Hello Julien,

thank you for verifying the commits!

> - Can you fix the licensing problems?

Corrected license to be Apache v2: https://github.com/shyrkov/izpack/commit/afdf422a2b807493d4f69c0530e2fa096384a6ce

> - Would you merge/port the changes to the master branch?

As we are not using the master branch (IzPack 5.0.x), it will be a bit tricky: I am not familiar with the IzPack 5 code and cannot test my changes (in order to not break something).
We use IzPack 4.3.x in our product (http://www.jahia.com/) and are really happy with the amount of features and simplicity IzPack provides. But due to lifecycle rules for the product, we can switch to IzPack 5.0.x only in the next release. The development will start in approx. two month. So, before that I doubt I will be able to upport our patches into 5.0.x master without introducing side-effects :-)

> - Would you update / contribute to the (new) documentation? http://docs.codehaus.org/display/IZPACK/User+documentation

I would with pleasure provide any changes/description for the submitted patches. But I am not sure about the way it should be done.
I see the 4.3.x documentation (at least, what has been imported) at http://docs.codehaus.org/display/IZPACK/Previous+Documentation but a warning says, it should not be modified.
Please, give me an advice and guidance on how I should proceed with documentation changes.

Thank you in advance!

Kind regards
Sergiy

**Comment by jponge** [comment]
Hi Sergiy,

> Corrected license to be Apache v2: https://github.com/shyrkov/izpack/commit/afdf422a2b807493d4f69c0530e2fa096384a6ce
> 
> Ok great. Could you please close the initial pull request and make me another one that also incudes that commit?
> As we are not using the master branch (IzPack 5.0.x), it will be a bit tricky: I am not familiar with the IzPack 5 code and cannot test my changes (in order to not break something).
> We use IzPack 4.3.x in our product (http://www.jahia.com/) and are really happy with the amount of features and simplicity IzPack provides. But due to lifecycle rules for the product, we can switch to IzPack 5.0.x only in the next release. The development will start in approx. two month. So, before that I doubt I will be able to upport our patches into 5.0.x master without introducing side-effects :-)
> 
> This makes me think that you could have an entry at http://izpack.org/we-love-izpack/ as contributors if you send me the materials (picture of the good size + text).

I would be very interested if you could port most of your features and fixes to the master branch within the next few months. As the code bases have vastly diverged this will probably require some manual cherry-picking, but I suppose that you understand that we need this for your patches to be useful to the project in the long term, not just for making a further release of the 4.3 branch.

Can I have your informal agreement to pursue this effort when you will have the time?

> would with pleasure provide any changes/description for the submitted patches. But I am not sure about the way it should be done.
> I see the 4.3.x documentation (at least, what has been imported) at http://docs.codehaus.org/display/IZPACK/Previous+Documentation but a warning says, it should not be modified.
> Please, give me an advice and guidance on how I should proceed with documentation changes.
> The previous documentation is indeed there as an old reference. It is an incomplete conversion of the old one. Otherwise you have the "User+Documentation" root page and a subtree starting from it. I cannot give you a very precise guidance for that apart that you should "organize your content in the way you would expect to see it as a user". A start is better than nothing, and because this is a wiki, we can still improve later on if need be.

Cheers

**Comment by jponge** [comment]
Oh sorry I did not see that the commits where added to the pull request, so please discard my first paragraph ;-)

**Comment by jponge** [comment]
I am sending an email to the other IzPack developers asking for reviews before we merge it in the next few days.

**Pull request merged**